core/src/main/java/com/novoda/noplayer/player/PlayerBuilder.java

```diff
@@ -0,0 +1,77 @@
+package com.novoda.noplayer.player;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+
+import com.novoda.noplayer.Player;
+import com.novoda.noplayer.drm.DownloadedModularDrm;
+import com.novoda.noplayer.drm.DrmHandler;
+import com.novoda.noplayer.drm.DrmType;
+import com.novoda.noplayer.drm.StreamingModularDrm;
+import com.novoda.noplayer.drm.provision.ProvisionExecutorCreator;
+import com.novoda.noplayer.exoplayer.NoPlayerExoPlayerCreator;
+import com.novoda.noplayer.exoplayer.drm.DrmSessionCreatorFactory;
+import com.novoda.noplayer.mediaplayer.NoPlayerMediaPlayerCreator;
+import com.novoda.utils.AndroidDeviceVersion;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class PlayerBuilder {
+
+    private DrmType drmType = DrmType.NONE;
+    private DrmHandler drmHandler = DrmHandler.NO_DRM;
+    private PrioritizedPlayerTypes prioritizedPlayerTypes = PrioritizedPlayerTypes.prioritizeExoPlayer();
+    private boolean downgradeSecureDecoder = false;
+
+    public PlayerBuilder withWidevineClassicDrm() {
+        return withDrm(DrmType.WIDEVINE_CLASSIC, DrmHandler.NO_DRM);
+    }
+
+    public PlayerBuilder withWidevineModularStreamingDrm(StreamingModularDrm streamingModularDrm) {
+        return withDrm(DrmType.WIDEVINE_MODULAR_STREAM, streamingModularDrm);
+    }
+
+    public PlayerBuilder withWidevineModularDownloadDrm(DownloadedModularDrm downloadedModularDrm) {
+        return withDrm(DrmType.WIDEVINE_MODULAR_DOWNLOAD, downloadedModularDrm);
+    }
+
+    public PlayerBuilder withDrm(DrmType drmType, DrmHandler drmHandler) {
+        this.drmType = drmType;
+        this.drmHandler = drmHandler;
+        return this;
+    }
+
+    public PlayerBuilder withPriority(PlayerType playerType, PlayerType... playerTypes) {
+        List<PlayerType> types = new ArrayList<>();
+        types.add(playerType);
+        types.addAll(Arrays.asList(playerTypes));
+        prioritizedPlayerTypes = new PrioritizedPlayerTypes(types);
+        return this;
+    }
+
+    public PlayerBuilder withDowngradedSecureDecoder() {
+        downgradeSecureDecoder = true;
+        return this;
+    }
+
+    public Player build(Context context) {
```

Comment:
Can `NoPlayerCreator` be properly used by the client? if it doesn't make sense for the client to do so, we might have to flatten the packages... java, libraries and packages are not friends, it really sucks :S